src/analysis/ASTnodefunctions.ts

```diff
@@ -29,120 +27,218 @@ import { FieldWeight, TypeWeightObject, Variables } from '../@types/buildTypeWei
  *
  */
 
-export function fieldNode(
-    node: FieldNode,
-    typeWeights: TypeWeightObject,
-    variables: Variables,
-    parentName: string
-): number {
-    let complexity = 0;
-    // 'resolvedTypeName' is the name of the Schema Type that this field resolves to
-    const resolvedTypeName =
-        node.name.value in typeWeights
-            ? node.name.value
-            : typeWeights[parentName].fields[node.name.value]?.resolveTo || null;
-
-    if (resolvedTypeName) {
+class ASTParser {
```

Comment:
This class business is real improvement over passing the type weights and variables through everything. Thanks for implementing this.